Apps/Playground/UWP/App.cpp

```diff
@@ -220,7 +227,21 @@ void App::OnPointerPressed(CoreWindow^, PointerEventArgs^ args)
     if (m_nativeInput != nullptr)
     {
         const auto& point = args->CurrentPoint->RawPosition;
-        m_nativeInput->MouseDown(0, static_cast<int>(point.X), static_cast<int>(point.Y));
+        if (!m_leftPressed && args->CurrentPoint->Properties->IsLeftButtonPressed)
+        {
+            m_leftPressed = true;
+            m_nativeInput->MouseDown(Babylon::Plugins::NativeInput::LEFT_MOUSE_BUTTON_ID, static_cast<int>(point.X), static_cast<int>(point.Y));
+        }
+        if (!m_middlePressed && args->CurrentPoint->Properties->IsMiddleButtonPressed)
+        {
+            m_middlePressed = true;
+            m_nativeInput->MouseDown(Babylon::Plugins::NativeInput::MIDDLE_MOUSE_BUTTON_ID, static_cast<int>(point.X), static_cast<int>(point.Y));
+        }
+        if (!m_rightPressed && args->CurrentPoint->Properties->IsRightButtonPressed)
+        {
+            m_rightPressed = true;
+            m_nativeInput->MouseDown(Babylon::Plugins::NativeInput::RIGHT_MOUSE_BUTTON_ID, static_cast<int>(point.X), static_cast<int>(point.Y));
+        }
```

Comment:
I don't understand why you need to keep track of the state manually. Are the UWP events not correct? Can you do the same as Win32?

Reply:
Explained in previous comment